cli/flox/src/commands/mod.rs

```diff
@@ -524,6 +526,16 @@ pub enum EnvironmentSelect {
     Unspecified,
 }
 
+#[derive(Debug, Error)]
+pub enum EnvironmentSelectError {
+    #[error(transparent)]
+    Environment(#[from] EnvironmentError2),
+    #[error("Did not find an environment in the current directory.")]
+    EnvNotFoundInCurrentDirectory,
```

Comment:
suggestion: specific error for env not found in specified dir

I know this is mostly a copy of the internal error but it better describes the error states this can land in.
Cleaner IMO, but not necessarily blocking

```suggestion
    #[error("Did not find an environment in the current directory.")]
    EnvNotFoundInCurrentDirectory,
    #[error("Did not find an environment in '{0}'")]
    EnvNotFoundInSpecifiedDir(PathBuf)
```

Reply:
I think I'll leave as is so we don't copy the internal error. We can keep an eye on whether internal errors should bubble up